cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java

```diff
@@ -205,14 +205,30 @@ private boolean modelContainsRoot(Model model, Resource subject) {
   private void processContext(Context context, Model model) {
     if (!configuration.isContextEnabled()) return;
 
-    Map<String, String> prefixes = conversionProfile.getContext().getPrefixes();
+    Map<String, String> prefixes = mergePrefixMaps(
+            conversionProfile.getContext().getPrefixes(),
+            model.getNsPrefixMap()
+    );
     if (MapUtils.isEmpty(prefixes)) return;
 
     ObjectNode contextNode = context.jsonRoot.putObject("context");
     ObjectNode prefixNode = contextNode.putObject("prefix");
     prefixes.forEach(prefixNode::put);
   }
 
+  private Map<String, String> mergePrefixMaps(Map<String, String> map1, Map<String, String> map2) {
+    Map<String, String> result = new HashMap<>(map1);
+
+    result.putAll(
```

Comment:
not sure if creating a new map and then invoking putAll is good: it is more lines of code and more CPU cycles.